packages/core/src/index.spec.ts

```diff
@@ -220,3 +221,60 @@ describe('Context disposal', () => {
     });
   });
 });
+
+describe('Service instantiator interceptors', () => {
+  class BaseServiceTest extends BaseService {}
+
+  it('register new interceptor', () => {
+    //given
+    const app = new App();
+    const myMockFunction = jest.fn();
```

Comment:
Instead of a mock function can we make a transformation function here to ensure that the interceptor is actually applied to the value?

Reply:
I added additional test method that checks if the interceptor can modify the service that we want to get.
I left this one as is just to check if the interceptor is invoked, that way we check the statement in the test 'register new interceptor'